feat: add MFS tutorial (#200)
ProtoSchool/protoschool.github.io: src/tutorials/Mutable-File-System/02.md

## Working with files in ProtoSchool

> Here in our ProtoSchool tutorials, rather than using your machine itself, we're creating a new IPFS node for you in the browser each time you hit the "Submit" button in a lesson. Whenever you see `ipfs.someMethod()` in our lessons, `ipfs` is a variable that refers to your IPFS instance, also known as a node. The actions that you take only affect your own IPFS node, not nodes belonging to your peers.

Reviewer: > I'm not sure I understand this sentence...is it referring to the difference between running a daemon on your machine and running it in the browser?

Yeah, I think @terichadbourne was trying to differentiate between having to install IPFS and run a daemon in your terminal from running an on-demand disposable one in the browser.

> Here in our ProtoSchool tutorials, rather than using your machine itself, we're creating a new IPFS node for you in the browser each time you hit the "Submit" button in a lesson. Whenever you see `ipfs.someMethod()` in our lessons, `ipfs` is a variable that refers to your IPFS instance, also known as a node. The actions that you take only affect your own IPFS node, not nodes belonging to your peers.

Author: Yeah, so essentially you can do all of ProtoSchool without ever learning to install IPFS. So we need to call out that we're skipping a whole step... which I can't describe accurately because I'm never done it NOT in the browser. Want to propose something that sounds accurate to you while still explaining what we're doing in the background?

> We're creating your IPFS node behind the scenes so you can focus on the content of our lessons, but eventually you'll need to learn to host your own node locally. When you're ready to experiment, you can find instructions for [installing IPFS](https://docs.ipfs.io/introduction/install/) and [initializing your node](https://docs.ipfs.io/introduction/usage/) in our docs.

Author: I've never done it the other way, so perhaps you can help me describe it more accurately. Would it be "eventually you'll need to learn to IPFS and run a daemon in your terminal"?

> We're creating your IPFS node behind the scenes so you can focus on the content of our lessons, but eventually you'll need to learn to host your own node locally. When you're ready to experiment, you can find instructions for [installing IPFS](https://docs.ipfs.io/introduction/install/) and [initializing your node](https://docs.ipfs.io/introduction/usage/) in our docs.

Author: Let me know how this sounds to you @alanshaw. Does it address your concerns? 

```
Here in our ProtoSchool tutorials, we create a new IPFS node for you in the browser each time you hit the "Submit" button in a lesson. Whenever you see `ipfs.someMethod()` in our lessons, `ipfs` is a variable that refers to your IPFS instance, also known as a node. The actions that you take only affect your own IPFS node, not nodes belonging to your peers.

We're creating your IPFS node behind the scenes so you can focus on the content of our lessons, but eventually you'll need to learn to host your own node locally by installing IPFS and running a daemon in your terminal. When you're ready to experiment, you can find instructions for [installing IPFS](https://docs.ipfs.io/introduction/install/) and [initializing your node](https://docs.ipfs.io/introduction/usage/) in our docs.
```

As mentioned previously, methods associated with the Mutable File System are part of the Files API, so they'll take the format of `ipfs.files.someMethod()`. Let's take a look at a simple method you can start using even before you've added any files to your IPFS node.

## Exploring your IPFS node with `ipfs.files.stat`
When working with your IPFS node, you'll often want to check the status of a file or directory. You can do this with `ipfs.files.stat`, passing in the path you'd like to check on.

For example, to check the status of a directory called `stuff` located within our root directory ( `/` ), we could call the method like so:

> `await ipfs.files.stat('/stuff/')`

Reviewer 2: ```suggestion
`await ipfs.files.stat('/stuff')`
```

This method returns an object with some essential data about our file or directory:

* **hash** (a string with the cryptographic hash)
* **size** (an integer with the file or folder size in Bytes)
* **cumulativeSize** (an integer with the size of the DAGNodes making up the file in Bytes)
* **type** (a string that can be either `directory` or `file`)
* **blocks** (if type is `directory`, this is the number of files in the directory; if type is `file`, it's the number of blocks that make up the file)
* **withLocality** (a boolean to indicate if locality information are present)
* **local** (a boolean to indicate if the queried dag is fully present locally)
* **sizeLocal** (an integer indicating the cumulative size of the data present locally)

> It's important to note that you can `stat` your IPFS node even when you don't have anything in it yet. Even empty nodes have CIDs (hashes) that represent their contents.

Reviewer 2: ```suggestion
It's important to note that you can `stat` your IPFS node even when you don't have anything in it yet. Even empty nodes have a CID (hash).
```

Author: There are two separate issues here - one is the plural. If you prefer it without the plural then we can switch to "Even an empty node has a CID (hash)" so the subject is singular also. 

I'm assuming your objection to "that represent their contents" was that this empty node doesn't have any contents, which is precisely what I was trying to call out. I want to remind people that the CID is directly connected to the data - or lack thereof. 

Would this make sense to you? 
"It's important to note that you can `stat` your IPFS node even when you don't have anything in it yet. Even an empty node has a CID (hash) that represents its (non-existent) contents." (or "contents (or lack thereof)")

I'm open to better suggestions!
